src/pydvl/valuation/methods/_solve_least_core_problems.py

```diff
@@ -0,0 +1,336 @@
+from __future__ import annotations
+
+import logging
+import warnings
+from functools import partial
+from typing import List, NamedTuple, Sequence, Tuple
+
+import cvxpy as cp
+import numpy as np
+from joblib import Parallel, delayed
+from numpy.typing import NDArray
+
+from pydvl.utils import Status
+from pydvl.valuation.result import ValuationResult
+from pydvl.valuation.types import Sample
+from pydvl.valuation.utility.base import UtilityBase
+
+__all__ = [
+    "_solve_least_core_linear_program",
+    "_solve_egalitarian_least_core_quadratic_program",
+    "lc_solve_problem",
+    "lc_solve_problems",
+    "LeastCoreProblem",
+]
+
+logger = logging.getLogger(__name__)
+
+
+class LeastCoreProblem(NamedTuple):
+    utility_values: NDArray[np.float_]
+    A_lb: NDArray[np.float_]
+
+
+def lc_solve_problem(
```

Comment:
This function is mixing two levels of abstraction, on the one hand it is doing a lot of low level stuff to construct the conditions for the optimization problem and on the other hand, it calls other more low-level functions (_solve_...). This mixing of levels of abstractions makes it harder to read. Even if a user would not call this function directly, for the developers it is a more difficult task to grasp what is happening (actually the comment helps a lot and in combination with the log messages it seems to me that the related code should be encapsulated in other low-level functions with only one purpose).

Reply:
I created an issue about this and related improvements (#584)